### ISSUE TITLE
Solutions for Unable to Open HTML Analysis Report 分析报告html无法打开的处理办法

### ISSUE THREAD

**Issue by lishuangshuang0616** [MGI-tech-bioinformatics/DNBelab_C_Series_HT_scRNA-analysis-software#191]
Since last week, users whose browsers do not have cached jQuery may find that the HTML fails to open.
Reason: The jQuery path referenced in the HTML has been blocked.
Current Solution: Replace the jQuery URL in the HTML file.

<img width="1845" height="436" alt="Image" src="https://github.com/user-attachments/assets/51a6c5a2-c792-46a6-b558-1cbbf681ba11" />

Simply replace the jQuery reference with the following URL to resolve the issue:
```http://code.jquery.com/jquery-1.9.1.min.js```

To perform batch replacement of jQuery references in generated HTML files, you can directly use the sed command as follows:
```
sed -i 's|http://lib.sinaapp.com/js/jquery/1.9.1/jquery-1.9.1.min.js|http://code.jquery.com/jquery-1.9.1.min.js|g' *.html
```
For Template HTML in Analysis Software (dnbc4tools v2.1.3)
The template HTML files are located at:
```dnbc4tools2.1.3/lib/python/dnbc4tools/config/template/```

Run the same sed command in this directory to update jQuery in the template files:

```
cd dnbc4tools2.1.3/lib/python/dnbc4tools/config/template/  
sed -i 's|http://lib.sinaapp.com/js/jquery/1.9.1/jquery-1.9.1.min.js|http://code.jquery.com/jquery-1.9.1.min.js|g' *.html
```

The fixed version (v2.1.3) will be updated in the installation directory shortly.
The modified HTML templates with jQuery replacements will also be provided for direct replacement.

**Comment by lishuangshuang0616** [comment]
# 关于 HTML 报告无法正常打开的说明及处理方法

尊敬的用户：

您好！

近期我们收到部分用户反馈，分析软件生成的 HTML 报告在浏览器中无法正常显示，出现页面空白或内容缺失的情况。经技术排查，确认原因为 HTML 中引用的 jQuery 库地址已失效，导致脚本加载失败。

为保障您正常浏览分析结果，现将问题说明及处理建议整理如下：

## 一、问题说明

HTML 报告中引用了外部 jQuery 库地址：

```shell
http://lib.sinaapp.com/js/jquery/1.9.1/jquery-1.9.1.min.js
```

该资源目前无法访问，致使报告中依赖 jQuery 的功能无法加载，页面内容无法显示。

## 二、推荐替换方式

请将原有 jQuery 引用地址替换为可用的新地址：

```shell
http://code.jquery.com/jquery-1.9.1.min.js
```

### 手动修改方法

1. 使用文本编辑器打开 HTML 文件：

   - Windows 推荐使用：记事本、Notepad++、VS Code
   - macOS 推荐使用：文本编辑（请先设置为将html显示为html代码而不显示格式化文本）、Sublime Text、VS Code
   - Linux 推荐使用：vim、nano

2. 查找并替换原始 jQuery 路径：

   对应位置如下图所示（以 RNA 分析报告为例）：

   <img src="https://private-user-images.githubusercontent.com/49019821/471294899-51a6c5a2-c792-46a6-b558-1cbbf681ba11.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM2OTYzOTUsIm5iZiI6MTc1MzY5NjA5NSwicGF0aCI6Ii80OTAxOTgyMS80NzEyOTQ4OTktNTFhNmM1YTItYzc5Mi00NmE2LWI1NTgtMWNiYmY2ODFiYTExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzI4VDA5NDgxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJhMDVmODViMTQyZTViMmFmYTg2N2FmM2EzMDUzZGIyNGEyMTc3YmI5OGJiMjVhNTNlY2M0NTIzZGQ3OThhNGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZPDBw2lHTGl3jN2gIayyKeZVb1lnXkQvoqlQ3ldbWXo" alt="image-20250728151550920" style="zoom: 40%;" /> 



   将以下内容：

   ```shell
   http://lib.sinaapp.com/js/jquery/1.9.1/jquery-1.9.1.min.js
   ```

   替换为：

   ```shell
   http://code.jquery.com/jquery-1.9.1.min.js
   ```

3. 保存文件并重新用浏览器打开。

### 批量修改方法（适用于 Linux/macOS 用户）

若需对多个 HTML 文件进行批量替换，可使用以下终端命令：

```shell
sed -i 's|http://lib.sinaapp.com/js/jquery/1.9.1/jquery-1.9.1.min.js|http://code.jquery.com/jquery-1.9.1.min.js|g' *.html
```

说明：

- 请在包含 HTML 文件的目录下执行。

## 三、分析软件模板的更新位置

分析软件（dncb4tools v2.1.3）中的 HTML 报告模板位置为：

```shell
dnbc4tools2.1.3/lib/python/dnbc4tools/config/template/
```

您可在此目录下执行相同的 sed 命令，批量更新模板中的 jQuery 引用路径。

```shell
cd dnbc4tools2.1.3/lib/python/dnbc4tools/config/template/ 
sed -i 's|http://lib.sinaapp.com/js/jquery/1.9.1/jquery-1.9.1.min.js|http://code.jquery.com/jquery-1.9.1.min.js|g' *.html
```



## 四、后续安排

我们将于近期在软件安装包中更新模板文件，版本号保持不变（v2.1.3）。同时，也将提供已修复的 HTML 模板，供您直接替换使用。

如在操作过程中遇到问题，欢迎联系技术支持团队协助处理。

感谢您的理解与支持！

**Comment by hullv** [comment]
您好，

我根据以下命令替换掉了模板里的html文件后，重新生成了报告文件，请问为何报告还是打不开呢？除非去替换掉报告文件中路径才可以打开文件。sudo sed -i 's|http://lib.sinaapp.com/js/jquery/1.9.1/jquery-1.9.1.min.js|http://code.jquery.com/jquery-1.9.1.min.js|g' *.html

<img width="1226" height="221" alt="Image" src="https://github.com/user-attachments/assets/a31144ff-9d67-4e18-905c-9a7d0b8fe980" />

请问这个问题应该如何解决呢？

谢谢！

**Comment by lishuangshuang0616** [comment]
你可以看下这个模板里面是不是确实替换掉了。
如果替换掉了之后，生成出来的报告肯定也是替换掉了。

**Comment by hullv** [comment]
> 你可以看下这个模板里面是不是确实替换掉了。 如果替换掉了之后，生成出来的报告肯定也是替换掉了。
模板已经替换成功了，但即使重新生成报告后，新的报告也是打不开。

<img width="2403" height="189" alt="Image" src="https://github.com/user-attachments/assets/4ab1f224-38c1-40e2-abee-9945bb68a9ad" />

**Comment by lishuangshuang0616** [comment]
你能发一个你分析的日志吗，我怀疑是不是用的不是这个目录的模版

**Comment by hullv** [comment]
[log.zip](https://github.com/user-attachments/files/21542776/log.zip)

**Comment by lishuangshuang0616** [comment]
这个样本跑了这么久啊。你重新运行下原来的命令行加上--process report参数看看生成的报告是否替换了呢？

**Comment by hullv** [comment]
> 这个样本跑了这么久啊。你重新运行下原来的命令行加上--process report参数看看生成的报告是否替换了呢？

[2025-08-05-10-15-45.025-ABoxHeadless.exe-1128.log](https://github.com/user-attachments/files/21653675/2025-08-05-10-15-45.025-ABoxHeadless.exe-1128.log)

抱歉，回复晚了。我新运行了一个ATAC样本，报告还是打不开。

**Comment by lishuangshuang0616** [comment]
> > 这个样本跑了这么久啊。你重新运行下原来的命令行加上--process report参数看看生成的报告是否替换了呢？
> 
> [2025-08-05-10-15-45.025-ABoxHeadless.exe-1128.log](https://github.com/user-attachments/files/21653675/2025-08-05-10-15-45.025-ABoxHeadless.exe-1128.log)
> 
> 抱歉，回复晚了。我新运行了一个ATAC样本，报告还是打不开。

这个有点奇怪。如果实在不行，可以重新下载软件试试。新的安装包是改过的。
